lib/engine.js

```diff
@@ -122,14 +135,11 @@ exports.create = function (type, config) {
         context = createRenderContext(options);
         if (config.stream) {
             stream = dust.stream(name, context);
-            stream.on('error', reset);
-            stream.on('end', reset);
             callback(null, new River(stream));
             return;
         }
 
         dust.render(name, context, function () {
-            reset();
             callback.apply(undefined, arguments);
```

Comment:
Since `reset();` is removed, this can be simplified to `dust.render(name, context, callback);`